### ISSUE TITLE
Ability adjust the amount of rooms shown when when room list is collapsed

### ISSUE THREAD

**Issue by wapsi** [vector-im/element-web#24286]
#### What would you like to do?
It would be nice, if the amount of rooms shown in room list when the list is collapsed can be adjusted from the settings.

#### Why would you like to do it?
When I'm using Element on my 2K resolution for example, there is enough space to show more rooms than the current, hard-coded amount: 8 rooms is. I've modified it to be 12 for example, and it is better than 8 in my case:

![image](https://user-images.githubusercontent.com/11405096/213931901-9477280d-02a6-4d5e-b567-22e61b46217c.png)

#### How would you like to achieve it?
Now the amount is hard-coded in defaultVisibleTiles, in matrix-react-sdk, and it's not configurable: https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/stores/room-list/ListLayout.ts
```
    public get defaultVisibleTiles(): number {
        // This number is what "feels right", and mostly subject to design's opinion.
        return 8;
    }
```
In my opinion, that value should be configurable variable in the settings.

### Have you considered any alternatives?

The hard-coded amount has been changed at least once: from 5 rooms to 8, here is the commit: https://github.com/matrix-org/matrix-react-sdk/pull/5821/commits/3f33060cddb150a3b17cf381d51ccaa036581ebd But the ultimate solution would be configurable setting for that.

**Comment by wapsi** [comment]
Yes, you're right! But it resets the size back to the original state (lists only 8 rooms again) if I expand the list and then collapse it again.

So, if that handle is used, then (in my opinion) it should remember the position where it was if the list is first expanded and then collapsed back again.

**Comment by t3chguy** [comment]
Fixed by new room list
